iceoryx_posh/CMakeLists.txt

```diff
@@ -297,27 +297,11 @@ setup_install_directories_and_export_package(
     INCLUDE_DIRECTORY include/
 )
 
-if(ROUDI_ENVIRONMENT OR BUILD_TEST)
-    #
-    ######### posh roudi environment ##########
-    #
-    iox_add_library(
-        STATIC
-        TARGET                  iceoryx_posh_testing
-        NAMESPACE               iceoryx_posh_testing
-        PROJECT_PREFIX          ${PREFIX}
-        BUILD_INTERFACE         ${CMAKE_CURRENT_SOURCE_DIR}/testing/include
-        INSTALL_INTERFACE       include/${PREFIX}
-        PRIVATE_LIBS            iceoryx_posh::iceoryx_posh
-                                iceoryx_hoofs::iceoryx_hoofs
-                                iceoryx_posh::iceoryx_posh_roudi
-                                iceoryx_hoofs_testing::iceoryx_hoofs_testing
-        FILES
-            testing/roudi_environment/runtime_test_interface.cpp
-            testing/roudi_environment/roudi_environment.cpp
-    )
+if(ROUDI_ENVIRONMENT OR BUILD_TESTING)
```

Comment:
This is currently kind of broken. The RouDiEnvironment needs iceoryx_hoofs_testing but that will only be build when `BUILD_TESTING` is true. Ideally the RouDiEnvironment should be build without depending on iceoryx_hoofs_testing which itself depends on gtest. The RouDiEnvironment is also interesting for language bindings test which use a different testing framework.

Nothing for this PR but we should create an issue to untangle the RouDiEnvironment from iceoryx_posh_testing.

Reply:
Issue created: https://github.com/eclipse-iceoryx/iceoryx/issues/1533